### PR TITLE
v3.0.0

### PR DESCRIPTION
**Pull request by zbrotherton**
# Pull Request Details

## Description

Full arm and intake functionality for both cones and cubes
Numerous autonomous routines added, including autonomous balancing
LED functionality added
REVEncoder support added to AbsoluteEncoder
DebugLog class added for improved logging

## Which does this affect?

ArmSubsystem
IntakeSubsystem
Auton
LEDSubsystems
AbsoluteEncoder
DebugLog

## Testing Done

Driveteam approval of all subsystems
DebugLogs function as intended
Auton testing wip

## Checklist

- [ ] Full system testing has been done on the robot to confirm functionality/hotfix
- [x] Pull request is coming from develop or a hotfix branch

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java
Comment (azvanderpas) [inline src/main/java/frc/robot/Robot.java:31]: Revert

File: src/main/java/frc/robot/RobotContainer.java
Comment (azvanderpas) [inline src/main/java/frc/robot/RobotContainer.java:145]: I wonder what the benefit of ProxyCommand wrapper is here?

.asProxy() is also available as a decorator

**Review by azvanderpas — DISMISSED** [review]

File: src/main/java/frc/robot/RobotContainer.java
Comment (zbrotherton2158) [reply src/main/java/frc/robot/RobotContainer.java:145]: ProxyCommand is used so that the switch statement logic runs each time the button is pressed, otherwise the method is called on initialization, and will use whatever the value of currentState was on initialization for the rest of the runtime

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/subsystems/IntakeSubsystem.java:54]: Having to know whether true = Cone, or not is confusing.  

Should we have setConeMode, setCubeMode with no parameters?
Or have a constant setMode(instakeSubsystem.coneMode)?

We can put this as a fix in a trello card, and get a new member to make the code changes and push to develop after release3 is merged, but we should still improve this.

Comment (zbrotherton2158) [reply src/main/java/frc/robot/subsystems/IntakeSubsystem.java:54]: Sure, put it in a card
Probably a good newcomer problem

**Review by azvanderpas — COMMENTED** [review]
Will approve once system testing is completed

**Comment by zbrotherton2158** [comment]
Auton works in simulation (As well as sim auton is expected to work...)
The current auton branch was created off of v3, we didn't have any issues with the routines used at waco
I don't know when we'll be able to test auton since we don't have any operating winch arms

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java
Comment (caseyjbrotherton) [reply src/main/java/frc/robot/subsystems/IntakeSubsystem.java:54]: Zane is making card

**Review by caseyjbrotherton — APPROVED** [review]

**Review by azvanderpas — APPROVED** [review]

**Pull request merged**
